app.py

```diff
@@ -122,3 +122,29 @@ def delete_post(post_id):
     db.session.delete(post)
     db.session.commit()
     return redirect(f"/users/{user.id}")
+
+@app.route("/tags")
+def show_tags():
+    tags = Tag.query.all()
+    return render_template("tags.html", tags=tags)
+
+@app.route("/tags/new")
+def new_tag_form():
+    return render_template("newtag.html")
+
+@app.route("/tags/new", methods=['POST'])
+def new_tag():
+    name = request.form['name']
+
+    tag = Tag(name=name)
+    db.session.add(tag)
+    db.session.commit()
+
+    return redirect("/tags")
+
+@app.route("/tags/<int:tag_id>")
+def tag(tag_id):
+    tag = Tag.query.get_or_404(tag_id)
+    post_id = tag.connect()
```

Comment:
Just a relationship, no need invocation,
```suggestion
    post_id = tag.connect
```